### PR TITLE
Update to Node v10.17.0

### PR DESCRIPTION
**Pull request by mcollina**
This is important because it adds:

* `Readable.from()`
*  Makes async iteration stable

### CODE REVIEW

File: lib/_stream_readable.js [hunk 1]
```diff
@@ -1075,9 +1100,64 @@ function endReadableNT(state, stream) {
...
+      if (!wState || wState.autoDestroy && wState.finished) {
+        stream.destroy();
+      }
+    }
   }
 }
 
+Readable.from = function (iterable, opts) {
+  var iterator;
+  if (iterable && iterable[Symbol.asyncIterator]) iterator = iterable[Symbol.asyncIterator]();else if (iterable && iterable[Symbol.iterator]) iterator = iterable[Symbol.iterator]();else throw new ERR_INVALID_ARG_TYPE('iterable', ['Iterable'], iterable);
```
[the remark this answers is not in the document]
Comment (mcollina) [reply lib/_stream_readable.js]: Good catch, fixed!

**Comment by mcollina** [comment]
@vweevers have you got an idea on why the browser tests are not running at all?

**Comment by vweevers** [comment]
Maybe because it's running on node 13? I haven't tested `airtap` on 13 yet.

**Comment by vweevers** [comment]
Looks like the same thing is happening on node 12.

@mcollina Does `npm run test-browser-local` work locally?

**Comment by mcollina** [comment]
@vweevers turns out the problem was in the code. They are passing locally now.

**Comment by mcollina** [comment]
@vweevers there seems to be an issue with IE11. The new `Readable.from()` internally uses async iteration, which we compile down to generators to support Node 6. However, generators are not available in IE11.

I see a some options:

1. Drop IE 11, but this would be better done in a semver-major release.
2. pluck out `Readable.from` from readable-stream
3. pluck out `Readable.from` from the browser build of readable-stream
4. do a browser build that ship regenerator or analog

I do not see many other options, but maybe there are.

I would really like to ship this update because it will remove the warning for async iteration. It’s about time.

**Comment by vweevers** [comment]
> Drop IE 11, but this would be better done in a semver-major release

+1. For what it's worth, feross's `buffer` will likely also drop IE11 in a next major, which will then propagate to `browserify`, `airtap`, etc.

> pluck out Readable.from from readable-stream

-1. Modularity is good, but in this case it's too surprising for users, because `readable-stream` is meant to mirror node streams. For a browserify build to "just work" it would have to detect `Readable.from` somehow and shim it separately. Perhaps it's worth a discussion in node. These static methods (`from`, `finished`, `pipeline`, `events.once`) would be easier to shim and tree-shake if they were separate modules in core.

> pluck out Readable.from from the browser build of readable-stream

If you mean disabling it (until the next major) then +1.

> do a browser build that ship regenerator or analog

-1. Although regenerator is small, it does add up if multiple modules include it, so I think polyfilling like this should be an application concern.

**Comment by mcollina** [comment]
Updated to Node 10.18.1, let's see how it goes.

**Comment by mcollina** [comment]
🎉  CI is passing. 

@vweevers @mafintosh would you like to take a final look?

**Comment by vweevers** [comment]
Approved it seconds before your comment :)

I suggest updating the PR title for the record.

**Pull request merged**